### PR TITLE
Changed every cost to loss, theta to theta hat

### PR DESCRIPTION
**Pull request by ashleychien**
Made changes according to this list:

Change theta star to theta_0
Change theta to \hat{theta_0}
Let’s use “average loss” and “loss” instead of “cost” and “loss”
Use “MSE is 15” instead of “MSE cost is 15”
Use “mean absolute loss is 12” instead of “mean absolute cost is 12”
Or “MAE is 12”
Add aside for f(x) in 10.1, don’t add f(x)

### CODE REVIEW

File: notebooks/ch11/gradient_basics.md [hunk 1]
```diff
@@ -23,12 +23,12 @@ pd.options.display.max_columns = 8
 
 ```python
 # HIDDEN
-def mse_cost(theta, y_vals):
+def mse(theta, y_vals):
```
Comment (SamLau95) [inline notebooks/ch11/gradient_basics.md]: Thanks for even updating the code! Appreciate the attention to detail :)

File: ch11/gradient_descent.md [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
-found the model that minimized the cost function.
+vary by table. Then, we decided to use the mean squared error loss function and
+found the model that minimized the loss function.
 
 We also found that there are simple expressions that minimize the MSE and the
-mean absolute cost functions: the mean and the median. However, as our models
-and cost functions become more complicated we will no longer be able to find
-useful algebraic expressions for the models that minimize the cost. For
-example, the Huber cost has useful properties but is difficult to differentiate
+mean absolute loss functions: the mean and the median. However, as our models
```
Comment (tjann) [inline ch11/gradient_descent.md]: It would be more consistent to say "mean absolute error loss functions"

File: notebooks/ch11/gradient_basics.md [hunk 3]
```diff
@@ -299,21 +299,21 @@ simple_minimize(huber_cost, tips['pcttip'], thetas)
 
 
 
-We can see that using the Huber cost gives us $ \theta = 15.5 $. We can now compare the minimizing $ \theta $ values for MSE cost, mean absolute cost, and Huber cost.
+We can see that using the Huber loss gives us $ \hat{\theta} = 15.5 $. We can now compare the minimizing $ \hat{\theta} $ values for MSE, mean absolute loss, and Huber loss.
```
Comment (tjann) [inline notebooks/ch11/gradient_basics.md]: mae

File: notebooks/ch11/gradient_basics.md [hunk 4]
```diff
@@ -299,21 +299,21 @@ simple_minimize(huber_cost, tips['pcttip'], thetas)
...
 
 
 ```python
-print(f"          MSE cost: theta = {tips['pcttip'].mean():.2f}")
-print(f"Mean Absolute cost: theta = {tips['pcttip'].median():.2f}")
-print(f"        Huber cost: theta = 15.50")
+print(f"               MSE: theta_hat = {tips['pcttip'].mean():.2f}")
+print(f"Mean Absolute loss: theta_hat = {tips['pcttip'].median():.2f}")
```
Comment (tjann) [inline notebooks/ch11/gradient_basics.md]: mae

File: notebooks/ch11/gradient_basics.md [hunk 5]
```diff
@@ -299,21 +299,21 @@ simple_minimize(huber_cost, tips['pcttip'], thetas)
...
+print(f"        Huber loss: theta_hat = 15.50")
 ```
 
-              MSE cost: theta = 16.08
-    Mean Absolute cost: theta = 15.48
-            Huber cost: theta = 15.50
-
+                   MSE: theta_hat = 16.08
+    Mean Absolute loss: theta_hat = 15.48
```
Comment (tjann) [inline notebooks/ch11/gradient_basics.md]: mae

File: notebooks/ch11/gradient_basics.md [hunk 6]
```diff
@@ -299,21 +299,21 @@ simple_minimize(huber_cost, tips['pcttip'], thetas)
...
+            Huber loss: theta_hat = 15.50
+    
 
-We can see that the Huber cost is closer to the mean absolute cost since it is less affected by the outliers on the right side of the tip percentage distribution:
+We can see that the Huber loss is closer to the mean absolute loss since it is less affected by the outliers on the right side of the tip percentage distribution:
```
Comment (tjann) [inline notebooks/ch11/gradient_basics.md]: mae

File: notebooks/ch11/gradient_convexity.md [hunk 7]
```diff
@@ -30,73 +30,72 @@ tips['pcttip'] = tips['tip'] / tips['total_bill'] * 100
```
Comment (tjann) [inline notebooks/ch11/gradient_convexity.md]: In this file, you use "mean absolute error", but I think you can abbreviate with MAE esp. when in parallel with MSE. We can ask @SamLau95  if you should reintroduce that MAE is mean absolute error. I do it in Ch10 already.

File: notebooks/ch11/gradient_stochastic.ipynb [hunk 8]
```diff
@@ -7,7 +7,8 @@
     "ExecuteTime": {
```
Comment (tjann) [inline notebooks/ch11/gradient_stochastic.ipynb]: To calculate the minimizing value of θ -> $\hat{\theta_0}$

Comment (tjann) [reply notebooks/ch11/gradient_stochastic.ipynb]: Overall, in this file, it looks theta was changed to \hat{theta} instead of \hat{theta_0}. The reason is that we're just looking at a population statistic, so we're estimating the theta_0.

File: notebooks/ch11/gradient_descent_define.md [hunk 9]
```diff
@@ -30,340 +30,315 @@ tips['pcttip'] = tips['tip'] / tips['total_bill'] * 100
...
     xs = np.arange(theta - 1, theta + 1, 0.05)
-    ys = cost_fn(theta, y_vals) + slope * (xs - theta)
+    ys = loss_fn(theta, y_vals) + slope * (xs - theta)
     plt.plot(xs, ys, zorder=3, c=sns.xkcd_rgb['green'], linestyle='--')
 ```
 
 ## Gradient Descent
 
-We are interested in creating a function that can minimize a cost function without forcing the user to predetermine which values of $ \theta $ to try. In other words, while the `simple_minimize` function has the following signature:
+We are interested in creating a function that can minimize a loss function without forcing the user to predetermine which values of $ \theta $ to try. In other words, while the `simple_minimize` function has the following signature:
```
Comment (tjann) [inline notebooks/ch11/gradient_descent_define.md]: without forcing the user to predetermine which values of θ -> theta to \hat{theta_0}

File: notebooks/ch11/gradient_descent_define.md [hunk 10]
```diff
@@ -379,41 +354,41 @@ np.mean([12.1, 12.8, 14.9, 16.3, 17.2])
...
-The Huber cost is:
+The Huber loss is:
 
 $$
-L_\delta(\theta, y) = \frac{1}{n} \sum_{i=1}^n \begin{cases}
-    \frac{1}{2}(y_i - \theta)^2 &  | y_i - \theta | \le \delta \\
-     \delta (|y_i - \theta| - \frac{1}{2} \delta ) & \text{otherwise}
+L_\delta(\hat{\theta}, y) = \frac{1}{n} \sum_{i=1}^n \begin{cases}
+    \frac{1}{2}(y_i - \hat{\theta})^2 &  | y_i - \hat{\theta} | \le \delta \\
+     \delta (|y_i - \hat{\theta}| - \frac{1}{2} \delta ) & \text{otherwise}
```
Comment (tjann) [inline notebooks/ch11/gradient_descent_define.md]: probably fine, but this equation is not parsing in nbviewer. just fyi in case we find out it doesnt show up on gitbook either

File: notebooks/ch11/gradient_descent_define.md [hunk 11]
```diff
@@ -1,6 +1,6 @@
```
Comment (tjann) [inline notebooks/ch11/gradient_descent_define.md]: Also for this file, thetas were changed to \hat{\theta} instead of \hat{\theta_0}

File: notebooks/ch11/gradient_convexity.md [hunk 12]
```diff
@@ -30,73 +30,72 @@ tips['pcttip'] = tips['tip'] / tips['total_bill'] * 100
...
 
 ## Convexity
 
-Gradient descent provides a general method for minimizing a function. As we observe for the Huber cost, gradient descent is especially useful when the function's minimum is difficult to find analytically. 
+Gradient descent provides a general method for minimizing a function. As we observe for the Huber loss, gradient descent is especially useful when the function's minimum is difficult to find analytically. 
 
 ## Gradient Descent Finds Local Minima
 
-Unfortunately, gradient descent does not always find the globally minimizing $ \theta $. Consider the following gradient descent run using an initial $ \theta = -21 $ on the cost function below.
+Unfortunately, gradient descent does not always find the globally minimizing $ \hat{\theta} $. Consider the following gradient descent run using an initial $ \hat{\theta}a = -21 $ on the loss function below.
```
Comment (tjann) [inline notebooks/ch11/gradient_convexity.md]: was the a after \hat{\theta} intentional?

File: notebooks/ch11/gradient_convexity.md [hunk 13]
```diff
@@ -148,91 +147,91 @@ plot_one_gd_iter(pts, -12.6, quartic_cost, grad_quartic_cost)
...
 
 ## Definition of Convexity
 
-For some functions, any local minimum is also a global minimum. This set of functions are called **convex functions** since they curve upward. For a constant model, the MSE, mean absolute error, and Huber cost are all convex.
+For some functions, any local minimum is also a global minimum. This set of functions are called **convex functions** since they curve upward. For a constant model, the MSE, mean absolute error, and Huber loss are all convex.
 
-With an appropriate learning rate, gradient descent finds the globally optimal $ \theta $ for convex cost functions. Because of this useful property, we prefer to fit our models using convex cost functions unless we have a good reason not to.
+With an appropriate learning rate, gradient descent finds the globally optimal $ \theta $ for convex loss functions. Because of this useful property, we prefer to fit our models using convex loss functions unless we have a good reason not to.
```
Comment (tjann) [inline notebooks/ch11/gradient_convexity.md]: finds the globally optimal θ -> \hat{\theta_0}

File: notebooks/ch11/gradient_convexity.md (hunk 7 above)
Comment (tjann) [reply notebooks/ch11/gradient_convexity.md]: Also for this file, thetas were changed to \hat{\theta} instead of \hat{\theta_0}

File: notebooks/ch11/gradient_basics.md [hunk 14]
```diff
@@ -39,81 +39,81 @@ def points_and_cost(y_vals, xlim, cost_fn):
...
 
 
 ![png](gradient_basics_files/gradient_basics_4_0.png)
 
 
-How might we write a program to automatically find the minimizing value of $ \theta $? The simplest method is to compute the cost for many values $ \theta $. Then, we can return the $ \theta $ value that resulted in the least cost.
+How might we write a program to automatically find the minimizing value of $ \hat{\theta} $? The simplest method is to compute the loss for many values $ \hat{\theta} $. Then, we can return the $ \hat{\theta} $ value that resulted in the least loss.
 
-We define a function called `simple_minimize` that takes in a cost function, an array of data points, and an array of $ \theta $ values to try.
+We define a function called `simple_minimize` that takes in a loss function, an array of data points, and an array of $ \theta $ values to try.
```
Comment (tjann) [inline notebooks/ch11/gradient_basics.md]: an array of θ values to try -> \hat{\theta_0}

File: notebooks/ch11/gradient_basics.md [hunk 15]
```diff
@@ -1,6 +1,6 @@
```
Comment (tjann) [inline notebooks/ch11/gradient_basics.md]: Also for this file, thetas were changed to \hat{\theta} instead of \hat{\theta_0}

**Comment by ashleychien** [comment]
@tjann Hi Tiff! I'd appreciate if you could look over these changes one more time (with Deb and Sam's updates from Friday night). Let me know when you're done with updating Ch10, and I'll look it over ^_^

File: notebooks/ch11/gradient_descent_define.md [hunk 16]
```diff
@@ -128,31 +128,31 @@ plot_tangent_on_loss(pts, 16.5, mse)
...
 $$
 
-Where $ \hat{\theta_t} $ is the current estimate and $ \hat{\theta_{t+1}} $ is the next estimate.
+Where $ \theta^t $ is the current estimate and $ \theta^{t+1} $ is the next estimate.
 
 For the MSE, we have:
 
 $$
 \begin{aligned}
 L(\hat{\theta_0}, y)
```
Comment (tjann) [inline notebooks/ch11/gradient_descent_define.md]: I think you forgot to change this one

File: notebooks/ch11/gradient_descent_define.md [hunk 17]
```diff
@@ -128,31 +128,31 @@ plot_tangent_on_loss(pts, 16.5, mse)
...
-&= -\frac{2}{n} \sum_{i = 1}^{n} (y_i - \hat{\theta_0}) \\
+&= \frac{1}{n} \sum_{i = 1}^{n}(y_i - \theta)^2\\
+\frac{\partial}{\partial \hat{\theta}} L(\theta, y)
+&= \frac{1}{n} \sum_{i = 1}^{n} -2(y_i - \theta) \\
+&= -\frac{2}{n} \sum_{i = 1}^{n} (y_i - \theta) \\
 \end{aligned}
 $$
 
-When $ \hat{\theta_t} = 12 $, we can compute $ -\frac{2}{n} \sum_{i = 1}^{n} (y_i - \hat{\theta_0}) = -5.2 $. Thus, $ \hat{\theta_{t+1}} = 12 - (-5.2) = 17.2 $.
+When $ \theta^t = 12 $, we can compute $ -\frac{2}{n} \sum_{i = 1}^{n} (y_i - \theta) = -5.2 $. Thus, $ \theta^{t+1} = 12 - (-5.2) = 17.2 $.
```
Comment (tjann) [inline notebooks/ch11/gradient_descent_define.md]: Here / earlier, do we want to use the time notation? I can see both sides :confused:

File: notebooks/ch11/gradient_descent_define.md [hunk 18]
```diff
@@ -168,13 +168,13 @@ plot_theta_on_loss(pts, 17.2, mse)
...
-Although $ \hat{\theta_0} $ went in the right direction, it ended up as far away from the minimum as it started. We can remedy this by multiplying the slope by a small constant before subtracting it from $ \hat{\theta_0}$. Our final update formula is:
+Although $ \theta $ went in the right direction, it ended up as far away from the minimum as it started. We can remedy this by multiplying the slope by a small constant before subtracting it from $ \theta$. Our final update formula is:
 
 $$
-\hat{\theta_{t+1}} = \hat{\theta_t} - \alpha \cdot \frac{\partial}{\partial \theta} L(\hat{\theta_t}, y)
+\theta^{t+1} = \theta^t - \alpha \cdot \frac{\partial}{\partial \theta} L(\theta^t, y)
 $$
 
-Where $ \alpha $ is a small constant. For example, if we set $ \alpha = 0.3 $ this is the new $ \hat{\theta_{t+1}} $:
+Where $ \alpha $ is a small constant. For example, if we set $ \alpha = 0.3 $ this is the new $ \theta^{t+1} $:
```
Comment (tjann) [inline notebooks/ch11/gradient_descent_define.md]: comma after 0.3

File: notebooks/ch11/gradient_descent_define.md [hunk 19]
```diff
@@ -251,24 +251,24 @@ plot_one_gd_iter(pts, 14.49, mse, grad_mse)
...
 To review notation:
 
-- $ \hat{\theta_t} $ is the current choice of $ \hat{\theta_0} $.
-- $ \hat{\theta_{t+1}} $ is the next choice of $ \hat{\theta_0} $.
-- $ \alpha $ is called the learning rate, usually set to a small constant. Sometimes it is useful to start with a larger $ \alpha $ and decrease it over time. If $ \alpha $ changes between iterations, we use the variable $ \alpha_t $ to mark that $ \alpha $ varies over time $ t $.
-- $ \nabla_\theta L(\hat{\theta_0}, y) $ is the partial derivative / gradient of the loss function at $ \hat{\theta_0} $.
+- $ \theta^t $ is the current choice of $ \theta $.
+- $ \theta_{t+1} $ is the next choice of $ \theta $.
+- $ \alpha $ is called the learning rate, usually set to a small constant. Sometimes it is useful to start with a larger $ \alpha $ and decrease it over time. If $ \alpha $ changes between iterations, we use the variable $ \alpha^t $ to mark that $ \alpha $ varies over time $ t $.
+- $ \nabla_\theta L(\theta, y) $ is the partial derivative / gradient of the loss function at $ \theta $.
```
Comment (tjann) [inline notebooks/ch11/gradient_descent_define.md]: I think we should sync this one up with the review in gradient_stochastic. I like how the one in gradient_stochastic mentions "current estimate of θ∗ at the tth iteration"

I think we can actually have the gradient bullet point match what is displayed above, and say it is the partial derivative / gradient of the loss function with respect to theta at theta^t

File: notebooks/ch11/gradient_descent_define.md (hunk 17 above)
Comment (SamLau95) [reply notebooks/ch11/gradient_descent_define.md]: I think the time notation makes sense here since we're performing an explicit update using the gradient descent algorithm.

As a sidenote, the `t` should be in parentheses: `\theta^{(t)}` and `\theta^{(t + 1)}`
